### ISSUE TITLE
.part files left behind, media download fails

### ISSUE THREAD

**Issue by shyd** [rarcega/instagram-scraper#161]
Hi,
there are some media files which fail to download. For example I scrape <https://www.instagram.com/lhtechnik/>, then some images fail, e.g. this one: <https://www.instagram.com/p/BOSRmePjESU/>. The json content for this post looks like this:

```json
    {
        "__typename": "GraphImage",
        "comments_disabled": false,
        "dimensions": {
            "height": 720,
            "width": 1080
        },
        "display_url": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/s1080x1080/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg",
        "edge_media_preview_like": {
            "count": 1444
        },
        "edge_media_to_caption": {
            "edges": [
                {
                    "node": {
                        "text": "We welcome Lufthansa's first #A350 in #Munich where our colleagues are looking forward to keeping good care of the newest family member! #keepyouflying"
                    }
                }
            ]
        },
        "edge_media_to_comment": {
            "count": 16
        },
        "id": "1410267043924296852",
        "is_video": false,
        "owner": {
            "id": "2296711840"
        },
        "shortcode": "BOSRmePjESU",
        "tags": [
            "Munich",
            "A350",
            "keepyouflying"
        ],
        "taken_at_timestamp": 1482336954,
        "thumbnail_resources": [
            {
                "config_height": 150,
                "config_width": 150,
                "src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/656615a4acbed6483913f68431a6fb8a/5ADDF25A/t51.2885-15/s150x150/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg"
            },
            {
                "config_height": 240,
                "config_width": 240,
                "src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/6192bf5e688fb8428752f1279f2a11c2/5B0187CC/t51.2885-15/s240x240/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg"
            },
            {
                "config_height": 320,
                "config_width": 320,
                "src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/4748000c3fb5df951eccf31b6b067fe0/5AFFE3CA/t51.2885-15/s320x320/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg"
            },
            {
                "config_height": 480,
                "config_width": 480,
                "src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/8521cef8677c97b497a22138ab1cb509/5AEA6179/t51.2885-15/s480x480/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg"
            },
            {
                "config_height": 640,
                "config_width": 640,
                "src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/4927c646bf32327b48a2ae2dccbe33b6/5AE86C34/t51.2885-15/s640x640/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg"
            }
        ],
        "thumbnail_src": "https://instagram.ftxl1-1.fna.fbcdn.net/vp/4927c646bf32327b48a2ae2dccbe33b6/5AE86C34/t51.2885-15/s640x640/e15/c180.0.720.720/15625063_1164138660342822_1366348496948428800_n.jpg",
        "urls": [
            "https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg"
        ]
    }
```

After the scrape is done, this media didn't get downloaded, there is just an empty file called `15625063_1164138660342822_1366348496948428800_n.jpg.part`.

The contents of `instagram-scraper.log` for this media are:

```
2018-01-21 11:09:10,597 - instagram_scraper.app - WARNING - Retry after exception StatusCodeException('Bad status code: 403',) on https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg
2018-01-21 11:09:15,737 - instagram_scraper.app - WARNING - Retry after exception StatusCodeException('Bad status code: 403',) on https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg
2018-01-21 11:09:25,881 - instagram_scraper.app - WARNING - Retry after exception StatusCodeException('Bad status code: 403',) on https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg
2018-01-21 11:09:46,041 - instagram_scraper.app - WARNING - Retry after exception StatusCodeException('Bad status code: 403',) on https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg
2018-01-21 11:10:26,217 - instagram_scraper.app - WARNING - Retry after exception StatusCodeException('Bad status code: 403',) on https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg
2018-01-21 11:11:46,435 - instagram_scraper.app - WARNING - Media at ['https://instagram.ftxl1-1.fna.fbcdn.net/vp/df9628897eacaaed69306ff7c4051415/5AE64AAE/t51.2885-15/e15/fr/15625063_1164138660342822_1366348496948428800_n.jpg'] generated an exception: Bad status code: 403
```

Opening the url in a browser leads to this output and error code 403 instead of the actual picture: `Invalid URL signature`

Maybe you could download `display_url` if `urls[0]` fails?

Cheers 🍻

**Comment by VasiliPupkin256** [comment]
It is duplicate for issue #157

**Comment by shyd** [comment]
Has been fixed by #168 
Thanks! 👍
